admin_app/src/app/playground/page.tsx

```diff
@@ -0,0 +1,209 @@
+      ? llmResponse
+      : `No response. Reason: "${response.debug_info.reason}". See JSON for details.`;
+
+    setMessages((prevMessages) => [
+      ...prevMessages,
+      {
+        dateTime: new Date().toISOString(),
+        type: "response",
+        content: responseText,
+        json: response,
+      },
+    ]);
+  };
+
+  const onSend = (queryText: string, queryType: QueryType) => {
+    if (queryText === "") {
+      return;
+    }
+
+    setLoading(true);
+
+    // TODO: Add loading component
+
+    if (currApiKey === null || currApiKey === "") {
+      setError("API Key not set. Please set the API key.");
+      setLoading(false);
+      return;
+    } else {
+      setMessages((prevMessages) => [
+        ...prevMessages,
+        {
+          dateTime: new Date().toISOString(),
+          type: "question",
+          content: queryText,
+        } as UserMessage,
+      ]);
+      if (queryType === "embeddings-search") {
+        apiCalls
+          .getEmbeddingsSearch(queryText, currApiKey)
+          .then((response) => {
+            processEmbeddingsSearchResponse(response);
+          })
+          .catch((error: Error) => {
+            setError("Embeddings search failed. See console for details.");
+            console.error(error);
+          })
+          .finally(() => {
+            setLoading(false);
+          });
+      } else if (queryType === "llm-response") {
+        apiCalls
+          .getLLMResponse(queryText, currApiKey)
+          .then((response) => {
+            processLLMSearchResponse(response);
+          })
+          .catch((error: Error) => {
+            setError("LLM Response failed. See console for details.");
+            console.error(error);
+          })
+          .finally(() => {
+            setLoading(false);
+          });
+      } else {
+        throw new Error("Invalid search option selected");
+      }
+    }
+  };
+
+  const [openDialog, setOpenDialog] = useState(false);
+
+  const [currApiKey, setCurrApiKey] = useState<string | null>(
+    typeof window !== "undefined" ? localStorage.getItem("apiToken") : null,
+  );
+  const handleDialogClose = () => {
+    setOpenDialog(false);
+  };
+
+  const handleDialogOpen = () => {
+    setOpenDialog(true);
+  };
+
+  const handleSaveToken = (token: string) => {
+    if (typeof window !== "undefined") {
+      localStorage.setItem("apiToken", token);
+      setCurrApiKey(token);
+    }
+
+    handleDialogClose();
+  };
+
+  const handleErrorClose = (
+    event?: React.SyntheticEvent | Event,
+    reason?: string,
+  ) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    setError(null);
+  };
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      if (!localStorage.getItem("apiToken")) {
+        handleDialogOpen();
+      }
+    }
+  }, []);
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      alignItems="center"
+      sx={{ height: "100vh", width: "100%", pb: 10 }}
+    >
```

Comment:
Making the full page white makes it look cleaner. Reference: ChatGPT's interface.


```suggestion
    <Box
      display="flex"
      flexDirection="column"
      alignItems="center"
      bgcolor="white"
      sx={{ height: "100vh", width: "100%", pb: 10 }}
    >
```